src/reducers/existingThoughtChange.js

```diff
@@ -53,6 +61,41 @@ export const existingThoughtChange = (state, { oldValue, newValue, context, show
     : thought
   )
 
+  const oldPath = thoughtsRankedLiveOld
+  const newPath = cursorNew
+
+  //cloning the state data and sorting the recent edit data by last updated field
+  const recentlyEditedUpdates = sortByLastUpdated(cloneDeep(state.recentlyEdited))
+
+  /** boolean to check if edited thought is already available inside the recently edited array.
+   *  if true it just needs to be updated and there is no need to append the incoming thought to the updated array
+  */
+  let isEditedContextAlreadyInState = false
+
+  recentlyEdited.forEach((data, index) => {
+    const path = data.path
+    const lastUpdated = data.lastUpdated
+
+    /*Replacing if the old path is already in the object*/
+    if (equalPath(path, oldPath)) {
+      isEditedContextAlreadyInState = true
+      recentlyEditedUpdates.splice(index, 1, { path: newPath, lastUpdated: timestamp() })
```

Comment:
Mutation is now allowed. (This was clearly stated in the [Contributing Guidelines](https://github.com/cybersemics/em/blob/dev/CONTRIBUTING.md#functional-programming))